### PR TITLE
Jenish/feature/iris-service-obc

### PR DESCRIPTION
**Pull request by JenishPatel99**
Includes iris service and sub-services definitions on obc side

### CODE REVIEW

File: ex2_hal/iris/hardware_interface/source/iris.c [hunk 1]
```diff
@@ -300,59 +286,55 @@ Iris_HAL_return iris_get_housekeeping(IRIS_Housekeeping *hk_data) {
...
-            case ERROR_STATE:
-            {
-                sys_log(INFO, "Iris returns NACK on housekeeping command");
-                return IRIS_HAL_ERROR;
+            } else {
+                controller_state = ERROR_STATE;
             }
+
+            // Transfer data from buffer to struct
+            // TODO: Verify Endianness and correct order of storage
```
Comment (Taylorr82) [inline ex2_hal/iris/hardware_interface/source/iris.c:310]: This is completed, right?

**Review by Taylorr82 — CHANGES_REQUESTED** [review]
Your while loops skip the log messages you put in ERROR_STATE and FINISH, we talked about this yesterday too, did you makes changes about that?

File: ex2_hal/iris/hardware_interface/source/iris.c [hunk 2]
```diff
@@ -49,26 +49,23 @@ Iris_HAL_return iris_take_pic() {
 
     while (controller_state != FINISH && controller_state != ERROR_STATE) {
```
[the remark this answers is not in the document]
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris.c]: Changes made :+1:

File: ex2_hal/iris/hardware_interface/include/iris.h [hunk 3]
```diff
@@ -12,20 +12,14 @@
  * GNU General Public License for more details.
  */
 
- /**
```
[the remark this answers is not in the document]
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/include/iris.h:15]: Should also check all of my files are properly named and commented. Thanks !

File: ex2_hal/iris/hardware_interface/source/iris.c (hunk 1 above)
Comment (JenishPatel99) [reply ex2_hal/iris/hardware_interface/source/iris.c]: Yes

**Review by Taylorr82 — APPROVED** [review]

**Pull request merged**